### ISSUE TITLE
no-computed-properties-in-native-classes fails with multiple class definitions

### ISSUE THREAD

**Issue by tylerbecks** [ember-cli/eslint-plugin-ember#1267]
If there are 2 classes defined in a file, one with the @classic decorator and one without, this rule will throw even if the class without the classic decorator doesn't have any computed properties. See example:

This fails (some code removed for brevity)
```
import classic from 'ember-classic-decorator';
// 👇  ❌  THE FOLLOWING LINE IS FAILING ESLINT
import { readOnly } from '@ember/object/computed';
import Component from '@ember/component';
// 👇  ❌  THE FOLLOWING LINE IS FAILING ESLINT
import { setProperties, get, getProperties, computed } from '@ember/object';
import layout from 'reports/templates/components/jobs-page/jobs-page-graph';

// 👇  HERE is the class that doesn't need the classic decorator. Removing it fixes the error.
class InchartsTooltip extends Tooltip {
  constructor(params, valueFormatter) {
    super(params);
    this.valueFormatter = ({ y }) => valueFormatter(y);
  }
}

//  👇  HERE is the class with the computed properties, appropriately marked with the classic decorator 
@classic
@templateLayout(layout)
export default class JobsPageGraph extends Component {
  @readOnly('tabName')
  'data-test-jobs-page-line-graph';

  @(computed('tabName').readOnly())
  get displayStrings() {
    const seriesName = get(this, 'tabName');

    return get(this, 'displayStringsMap')[seriesName] || {};
  }
}

```

**Comment by bmish** [comment]
Yeah I think this rule needs to be rewritten to take multiple classes into account. We have other rules that handle multiple classes properly.

**Comment by aggmoulik** [comment]
Hey @bmish Can you let me know the rules which handles multiple classes ?

**Comment by bmish** [comment]
Search the codebase for `currentClass`, `currentEmberClass`, or `currentEmberComponent` to see some implementations.

In particular, here's an example from the `no-unused-services` rule that uses a stack to keep track of the current class.

```js
return {
  ClassDeclaration(node) {
    // Native JS class
    classStack.push(node);
  },
  CallExpression(node) {
    if (emberUtils.isAnyEmberCoreModule(context, node)) {
      // Classic class
      classStack.push(node);
    }
  },
  'ClassDeclaration:exit'(node) {
    if (classStack.peek() === node) {
      // Leaving current (native) class.
    }
  },
  'CallExpression:exit'(node) {
    if (classStack.peek() === node) {
      // Leaving current (classic) class.
    }
  },
};
```
